services/pull/update.go

```diff
@@ -104,11 +104,25 @@ func IsUserAllowedToUpdate(pull *models.PullRequest, user *user_model.User) (mer
 		return false, false, nil
 	}
 
+	baseRepoPerm, err := models.GetUserRepoPermission(pull.BaseRepo, user)
+	if err != nil {
+		return false, false, err
+	}
+
 	mergeAllowed, err = IsUserAllowedToMerge(pr, headRepoPerm, user)
 	if err != nil {
 		return false, false, err
 	}
 
+	if pull.AllowEditsFromMaintainers {
```

Comment:
Should a maintainer which has just access to the PR's branch be able to merge that PR as well, I'm not sure. It only states that maintainers are able to edit the files of the PR.

Reply:
You can't merge PRs with the permission given with this option. (Everything else should work, but pull merging requires full code write permissions)

Reply:
Tested and I can't merge